earth_enterprise/src/common/khCache.h

```diff
@@ -80,6 +81,7 @@ class khCache {
   bool verbose;
 #endif
   uint numItems;
+  std::recursive_mutex mtx;
```

Comment:
I made this recursive instead of the regular mutex because both the public `Add` and `Remove` methods call `Prune` which is also public.

The cache definitely needs some form of protection on public methods when we move to use anything in parallel.